File: ohara-kafka-data/src/main/scala/com/island/ohara/kafka/RowSerializer.scala
package com.island.ohara.kafka

import java.util

import com.island.ohara.core.Row
import com.island.ohara.serialization.RowWriter
import org.apache.kafka.common.serialization.Serializer

/**
  * Used to convert ohara row to byte array. It is a private class since ohara producer will instantiate one and pass it to
  * kafka producer. Hence, no dynamical call will happen in kafka producer. The access exception won't be caused.
  */
class RowSerializer extends Serializer[Row] {
  override def configure(configs: util.Map[String, _], isKey: Boolean): Unit = {
    // do nothing
  }

  override def serialize(topic: String, data: Row): Array[Byte] = RowWriter.toBytes(data)

  override def close(): Unit = {
    // do nothing
  }
}

File: ohara-streams/src/main/java/com/island/ohara/ostreams/RowSerializer.java
package com.island.ohara.ostreams;

import com.island.ohara.common.data.Row;
import com.island.ohara.common.data.Serializer;
import java.util.Map;

public class RowSerializer implements org.apache.kafka.common.serialization.Serializer<Row> {

  @Override
  public void configure(Map<String, ?> configs, boolean isKey) {}

  @Override
  public byte[] serialize(String topic, Row data) {
    if (data == null) return null;
    else return Serializer.ROW.to(data);
  }

  @Override
  public void close() {}
}
